WordPress/src/main/java/org/wordpress/android/ui/posts/PostListEventListener.kt

```diff
@@ -83,63 +95,73 @@ class PostListEventListener(
      */
     @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
     private fun onDestroy() {
+        job.cancel()
         lifecycle.removeObserver(this)
         dispatcher.unregister(this)
         EventBus.getDefault().unregister(this)
     }
 
+    /**
+     * Has lower priority than the PostUploadHandler and UploadService, which ensures that they already processed this
+     * OnPostChanged event. This means we can safely rely on their internal state being up to date.
+     */
     @Suppress("unused")
-    @Subscribe(threadMode = BACKGROUND)
+    @Subscribe(threadMode = MAIN, priority = 5)
     fun onPostChanged(event: OnPostChanged) {
-        when (event.causeOfChange) {
```

Comment:
I just wrapped the body of the method with `launch`. I'm not sure why GitHub's diff looks so complicated.

Reply:
You can use the ` Hide whitespace changes` setting at the top of the page (under Settings icon) to make it easier to see the changes.